core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -900,24 +900,41 @@ public void setColumnComment(Session session, TableHandle tableHandle, ColumnHan
     public void renameColumn(Session session, TableHandle tableHandle, ColumnHandle source, String target)
     {
         CatalogName catalogName = tableHandle.getCatalogName();
-        ConnectorMetadata metadata = getMetadataForWrite(session, catalogName);
-        metadata.renameColumn(session.toConnectorSession(catalogName), tableHandle.getConnectorHandle(), source, target.toLowerCase(ENGLISH));
+        CatalogMetadata catalogMetadata = getCatalogMetadataForWrite(session, catalogName);
+        ConnectorMetadata metadata = catalogMetadata.getMetadata();
+        Optional<CatalogSchemaTableName> tableName = getTableNameIfSystemSecurity(session, catalogMetadata, tableHandle);
+        ConnectorSession connectorSession = session.toConnectorSession(catalogName);
+        metadata.renameColumn(connectorSession, tableHandle.getConnectorHandle(), source, target.toLowerCase(ENGLISH));
+        tableName.ifPresent(name -> systemSecurityMetadata.columnRenamed(
+                session,
+                name,
+                metadata.getColumnMetadata(connectorSession, tableHandle.getConnectorHandle(), source).getName(),
```

Comment:
I guess this should be similar to how we handle during `delete`. If Connector's Metadata Impl is based on `ColumnHandleImpl#getName` then it might fail.